drivers: mcp230xx: fix for multiple instances

arbrauns opened the pull request:
Parts of the macro used `inst`, others `n` for the instance number; only `n` was actually defined (so `inst` expanded to the empty string, causing name errors with more than one instance).

Unify everything to `inst`.

arbrauns committed:
drivers: mcp230xx: formatting

It's a formatting cleanup. Not much to explain here, but CI requires a body
text.

Signed-off-by: Armin Brauns <armin.brauns@embedded-solutions.at>

arbrauns committed:
drivers: mcp230xx: fix for multiple instances

Parts of the macro used `inst`, others `n` for the instance number; only
`n` was actually defined (so `inst` expanded to the empty string, causing
name errors with more than one instance).

Unify everything to `inst`.

Signed-off-by: Armin Brauns <armin.brauns@embedded-solutions.at>

bbilas commented on drivers/gpio/gpio_mcp230xx.c:
```diff
@@ -77,30 +77,28 @@ static int mcp230xx_bus_is_ready(const struct device *dev)
-	DEVICE_DT_INST_DEFINE(n, gpio_mcp23xxx_init, NULL, &mcp230xx_##inst##_drvdata,             \
-			      &mcp230xx_##inst##_config, POST_KERNEL,                              \
-			      CONFIG_GPIO_MCP230XX_INIT_PRIORITY, &gpio_mcp23xxx_api_table);
+	static struct mcp23xxx_config mcp230xx_##inst##_config = {                                 \
```
```suggestion
	static const struct mcp23xxx_config mcp230xx_##inst##_config = {                                 \
```

arbrauns replied:
Done.

arbrauns committed:
drivers: mcp230xx: make config struct `const`

`struct device::config` is a `const void *`, so this struct does not need
to be mutable.

Signed-off-by: Armin Brauns <armin.brauns@embedded-solutions.at>

arbrauns commented:
Compliance checks are now somehow failing on a commit 63d48dda11 that does not even exist on my end: https://github.com/zephyrproject-rtos/zephyr/actions/runs/3459547698/jobs/5775051685

bbilas commented:
> Compliance checks are now somehow failing on a commit 63d48dda11 that does not even exist on my end: https://github.com/zephyrproject-rtos/zephyr/actions/runs/3459547698/jobs/5775051685

It seems that commit sha 68421a9 does not contain a body message. You have to fill it.

PS I accidentally clicked "update branch" here, can anyone with permissions rebase it and force push?
